### ISSUE TITLE
Enable trading food

### ISSUE THREAD

**Issue by LynxAbraxas** [civctp2/civctp2#196]
There already seems to be quite a bit of game logic to enable trading food in addition to goods, see e.g. all `ROUTE_TYPE_FOOD` as in:

https://github.com/civctp2/civctp2/blob/08b9f1bdb09e67bf26b70d976fe358c02ec5c36b/ctp2_code/gs/gameobj/CityData.cpp#L4275-L4290

https://github.com/civctp2/civctp2/blob/08b9f1bdb09e67bf26b70d976fe358c02ec5c36b/ctp2_code/gs/gameobj/CityData.cpp#L4731-L4736

This would be very interesting to enable in conjuction with #188 (only giving a means of providing food through goods) but would probably need quite some work as it would need some kind of window or tab in the trade manager to set up such routes and code to teach the AI to do so (and to pirate such routes).

**Comment by MartinGuehmann** [comment]
(and to pirate such routes).

Well, I would assume that you would just pirate these routes as the other ones, unless you wanna give some priority to those.

**Comment by LynxAbraxas** [comment]
True, but currently I have the impression that the AI does only decide to pirate some route not taking any of the resulting effects into account. Even though the game already contained the route got cancelled message, I do not have the impression that the AI was reacting on that fact. With #188 it is now possible to weaken specific cites by pirating routes leading to them, before it was just loss of gold to the whole civ/player.
